paasta_tools/cli/cmds/status.py

```diff
@@ -347,81 +324,72 @@ def verify_instances(args_instances, service, clusters):
     return unverified_instances
 
 
-def paasta_args_mixer(args, service):
-    if args.deploy_group is None:
-        cluster_whitelist = args.clusters.split(",") if args.clusters is not None else []
-        if args.instances is not None:
-            instance_whitelist = verify_instances(args.instances, service, cluster_whitelist)
-        else:
-            instance_whitelist = []
+def get_filters(args):
+    if not any([args.clusters, args.instances, args.owner, args.deploy_group]):
+        service = figure_out_service_name(args, soa_dir=args.soa_dir)
```

Comment:
I see.

When args.service is None, that line is not executed.
This leads to a bit different behavior than we have now when it tries to guess service name (https://github.com/Yelp/paasta/blob/master/paasta_tools/cli/utils.py#L674).

```
$ cd ../yelpsoa-config/yelp-main
$ paasta status -c norcal-stagef
```

Expected result: it shows status of all yelp-main instances in norcal-stagef.
Actual result: paasta status iterates over all services.